Lib/unittest/mock.py

```diff
@@ -862,7 +862,9 @@ def assert_has_calls(self, calls, any_order=False):
                 not_found.append(kall)
         if not_found:
             raise AssertionError(
-                '%r not all found in call list' % (tuple(not_found),)
+                '%r does not contain all of %r in its call list, '
+                'found %r instead' % (self._mock_name or 'mock',
+                                      tuple(not_found), all_calls)
```

Comment:
Not sure about `tuple` here, given that both `not_found` and `all_calls` are lists as evident from lines 855, 857.

Reply:
While this change preserved the behavior of the existing code.  I don't really think it matters, the error message will be understandable regardless.